test/functional/vimscript/system_spec.lua

```diff
@@ -190,6 +190,23 @@ describe('system()', function()
       eq(iswin() and '??\n' or 'ああ\n', eval([[system('Write-Output "ああ"')]]))
     end)
 
+    it('works with powershell for filter and redirection (#16271)', function()
+      if not helpers.has_powershell() then
+        pending("not tested; powershell was not found", function() end)
+        return
+      end
+      local tempfile = helpers.tmpname()
+      helpers.set_shell_powershell()
+      screen:try_resize(500, 8)
+      feed(':edit ' .. tempfile .. '<cr>')
+      feed('i3<cr>1<cr>2<esc>:write<cr>')
```

Comment:
This will leave behind a `tempfile`. Is writing to a file necessary in this test?

Reply:
It's not necessary. Technically I can call `:%!sort` on an empty file. But I'm running into a problem here. For some reason, that same test, just without ".*3 lines filtered" is causing the running of test to get stuck (I have to ctrl-c out of it).

```lua
    it('works with powershell for filter and redirection (#16271)', function()
      -- same thing as before
      screen:expect{
        any=[[Executing command: "Start%-Process sort %-RedirectStandardInput .* %-RedirectStandardOutput .* %-NoNewWindow %-Wait]]
      }
      feed('<cr>')
    end)
```

The output gets stuck here:

```
enan@cirdan:~/projects/neovim$ make functionaltest TEST_FILE=test/functional/vimscript/system_spec.lua
mkdir -p build
touch build/.ran-third-party-cmake
ninja  -C .deps
ninja: Entering directory `.deps'
[3/3] cd /home/enan/projects/neovim/.deps && /usr/bin/cmake -E touch .third-party
ninja  -C build
ninja: Entering directory `build'
[0/2] Re-checking globbed directories...
[1/6] cd /home/enan/projects/neovim && /home/...build/config/auto/versiondef_git.h v0.8.0-dev
ninja  -C build functionaltest
ninja: Entering directory `build'
[0/2] Re-checking globbed directories...
[1/2] cd /home/enan/projects/neovim/build && ...ome/enan/projects/neovim/cmake/RunTests.cmake
[----------] Global test environment setup.
[----------] Running tests from test/functional/vimscript/system_spec.lua
[ RUN      ] system() command passed as a List throws error if cmd[0] is not executable: 3.61 ms OK
[ RUN      ] system() command passed as a List parameter validation does NOT modify v:shell_error: 5.28 ms OK
[ RUN      ] system() command passed as a List quotes arguments correctly #5280: 5.53 ms OK
[ RUN      ] system() command passed as a List calls executable in $PATH: 3.61 ms SKIP
[ RUN      ] system() command passed as a List does NOT run in shell: 4.20 ms OK
[ RUN      ] system() sets v:shell_error: 10.16 ms OK
[ RUN      ] system() executes shell function works with powershell w/ UTF-8 text (#13713): 518.15 ms OK
[ RUN      ] system() executes shell function works with powershell for filter and redirection (#16271): 11.12 ms OK
```
So this test runs okay, but then the rest gets stuck. I must be misunderstanding how screen:expect works.

Reply:
Not necessary. I can call `:%!sort` on an empty file (or `[No Name]` file). I've pushed the changed test.